### PR TITLE
[SofaSphFluid] Add sprite-based point render

### PR DESCRIPTION
**Pull request by fredroy**
Implementation of http://developer.download.nvidia.com/presentations/2010/gdc/Direct3D_Effects.pdf

ezgif-6-9b93f0eae158

Allows nice (?) renders without relying on surface reconstruction etc.
In this SPH simulation (10k particules) , the rendering takes <1% of the resources.

Also tweak a bit the default Z planes with QGLViewer (it has a weird behavior with narrow BBox scenes)

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by damienmarchal** [comment]
SofaAward ...
:arrow_right: https://www.sofa-framework.org/sofa-events/sofa-week-2019/#SOFA%20Awards

File: applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#include "OglFluidModel.inl"
```
Comment (epernod) [inline applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.cpp]: <SofaSphFluid/OglFluidModel.inl> ?

File: applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.h [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+#include <SofaSphFluid/config.h>
+
+#include <sofa/core/visual/VisualModel.h>
+#include <sofa/helper/gl/FrameBufferObject.h>
+#include <sofa/helper/gl/GLSLShader.h>
+#include <sofa/core/topology/BaseMeshTopology.h>
+#include <sofa/core/behavior/MechanicalState.h>
+#include <sofa/defaulttype/VecTypes.h>
+#include <SofaBaseTopology//TopologyData.inl>
```
Comment (epernod) [inline applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.h]: Don't think this include is needed

File: applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.inl [hunk 3]
```diff
@@ -0,0 +1,704 @@
+#pragma once
+
+#include "OglFluidModel.h"
```
Comment (epernod) [inline applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.inl]: same

File: applications/sofa/gui/qt/viewer/qgl/QtGLViewer.cpp [hunk 4]
```diff
@@ -138,8 +138,8 @@ QtGLViewer::QtGLViewer(QWidget* parent, const char* name, const unsigned int nbM
 
     setManipulatedFrame( new qglviewer::ManipulatedFrame() );
     //near and far plane are better placed
-    camera()->setZNearCoefficient(0.001);
-    camera()->setZClippingCoefficient(5);
+    camera()->setZNearCoefficient(0.005);
+    camera()->setZClippingCoefficient(2);
```
Comment (epernod) [inline applications/sofa/gui/qt/viewer/qgl/QtGLViewer.cpp:142]: better parameters?

**Comment by epernod** [comment]
Very cool, I just tested it. 
- But I see it is not possible to put black background?
- The fluid is not visible in front of another VisualModel, is it due to ZBuffer?

File: applications/sofa/gui/qt/viewer/qgl/QtGLViewer.cpp (hunk 4 above)
Comment (fredroy) [reply applications/sofa/gui/qt/viewer/qgl/QtGLViewer.cpp:142]: I just set the default parameters from QGLViewer documentation. (and the previous ones was messing up my scene :p )

**Comment by fredroy** [comment]
Those issues could be fixed when the OglFluidModel will be a VisualManager or something like that, to do multitexturing with the current FrameBuffer in Sofa.
This rendering is mostly a Proof of Concept so I shall upgrade it when I have time in the future ;)
Or if it is deemed necessary...

**Comment by epernod** [comment]
ezgif com-crop

Tested with the last changes in ParticleSource and ParticleSink ;)

**Pull request merged**